woocommerce/woocommerce-gutenberg-products-block
# Get preview data from Editor Context /

This PRs does three things, the first (which ended up being useless for this issue but still a better refactor) simply moves all previewData to be exposed from the `EditorContextProvider` instead of getting directly from `@woocommerce/resources-previews`.
So now we import `@woocommerce/resources-previews` inside the editor context and pass it to the provider and other hooks consume that context, however this didn't solve anything since we're still importing `useEditorContext` which lives with the other imports, so I did three solutions (I reverted the second and third in favor of the fourth)

Second solution:
- I aliased resource previews to `null-loader`, however this pattern won't be supported in webpack 5, and I can't seem to get the example in webpack docs working (aliasing it to null).
this solved the issue.

Third Solution:
- I separated context definition from the provider creation, so now when I import `useContextProvider` I don't have to import preview data with it, this solved the issue but I'm not happy with the naming or config (any ideas to refactor are welcome).

Forth Solution (suggested by @nerrad):
- do things as step one, but instead of passing the preview on the provider definition, pass it as a prop in the Cart/Edit or Checkout/Edit, much cleaner solution.


Fixes #2114



### How to test the changes in this Pull Request:

Cart block didn't use any previews, Checkout did
1. Test Checkout and Cart in the editor and see nothing is broken.
2. Test them in frontend.

for this to work, we will see a reduced bundle size in the bot comment here, I tested locally and preview is no longer imported.
